pandas/core/series.py

```diff
@@ -2061,18 +2061,24 @@ def map(self, arg, na_action=None):
         y : Series
             same index as caller
         """
-        values = self._values
-        if needs_i8_conversion(values.dtype):
-            boxer = i8_boxer(values)
-            values = lib.map_infer(values, boxer)
-
-        if na_action == 'ignore':
-            mask = isnull(values)
-
-            def map_f(values, f):
-                return lib.map_infer_mask(values, f, mask.view(np.uint8))
+        if is_extension_type(self.dtype):
+            values = self._values
+            if na_action is not None:
+                raise NotImplementedError
```

Comment:
tested in here.

https://github.com/pydata/pandas/pull/12532/files#diff-3c2759e1313d1a5f367b4b73020a18e7R302